numpy/distutils/unixccompiler.py

```diff
@@ -44,8 +44,9 @@ def UnixCCompiler__compile(self, obj, src, ext, cc_args, extra_postargs, pp_opts
             self.linker_so = llink_s.split() + opt.split()
 
     display = '%s: %s' % (os.path.basename(self.compiler_so[0]), src)
+    deps = ['-MMD', '-MF', obj + '.d']
```

Comment:
this deserves a comment explaining what's going on here. I could figure it out, but those are not flags one uses every day